lib/utility.js

```diff
@@ -43,13 +43,10 @@ async function loadLocalConf() {
 }
 
 export async function loadData(id) {
-	let path = '../data/' + id + '.js'
-
 	/* Check for config override file. */
 	const lconf = await loadLocalConf()
-	if (lconf && lconf.data_paths[id]) {
-		path = lconf.data_paths[id]
-	}
+	const path = lconf?.data_paths?.[id] ??
+		('../data/' + id + '.js')
```

Comment:
Two things:
1. The commit message does not reflect the changes done in the commit: There are no functional changes as far as I can tell. The commit mostly reduces naive checks and pre-emptive return values with new js [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and [Nullish Coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing). Which is fine, but the commit message should reflect as such.
2. I would really appreciate if we could start using [Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) instead of `"text" + obj + "text"`, this is extremely error-prone thanks to javascript's duck typing. :see_no_evil: 

<details>
<summary>Somewhat contrived String Concatenation Problem Example</summary>

```js
5 + 10 + "A"
```
yields `15A`.

While
```js
`${5}${10}A`
```
yields `510A`.
</details>

Reply:
1. The fix label is correct, imho.  Before the change, you could do a `touch .vitepress/local.js` and execution would crash, because the expectation is that if local.js exists that ALL configuration items must be in it.  This change makes every configuration item optional.  While technically only needed for the immediate markdown addition (so Pro will still build unless/until we commit the markdown extension there), we might as well add it to all config options since this should be the expected behavior.  (It's useful for testing as well, as you can create a local.js file with just the config item you want to overwrite from the default)  The javascript notation stuff is just to condense to a couple of lines, instead of the 5 or 6 separate if/else statement that would be needed otherwise for each function.
2. Sure.  Although I specifically have kept away from enforcing or implementing explicit coding rules for documentation, because at the end of the day this is not our core product.  It's just utility code used to build an ancillary part of the product (documentation).  We should broadly try to keep the same coding style for any addition, if just for code cleanliness reasons, but we aren't shipping this code for external use, for example.

Reply:
OK, I see now.